pandas/tests/window/test_timeseries_window.py

```diff
@@ -690,3 +705,34 @@ def test_rolling_cov_offset(self):
 
         expected2 = ss.rolling(3, min_periods=1).cov()
         tm.assert_series_equal(result, expected2)
+
+    def test_rolling_on_decreasing_index(self):
+        # GH-19248
+        index = [
+            Timestamp("20190101 09:00:00"),
+            Timestamp("20190101 09:00:02"),
+            Timestamp("20190101 09:00:03"),
+            Timestamp("20190101 09:00:05"),
+            Timestamp("20190101 09:00:06"),
+        ]
+
+        df = DataFrame({"column": [3, 4, 4, 2, 1]}, index=reversed(index))
+        result = df.rolling("2s").min()
+        expected = DataFrame(
+            {"column": [3.0, 3.0, 3.0, 2.0, 1.0]}, index=reversed(index)
+        )
+        tm.assert_frame_equal(result, expected)
+
+    def test_rolling_on_multi_index_level(self):
+        # GH-15584
+        df = DataFrame(
+            {"column": range(6)},
+            index=MultiIndex.from_product(
+                [date_range("20190101", periods=3), range(2)], names=["date", "seq"]
+            ),
+        )
+        result = df.rolling("10d", on=df.index.get_level_values("date")).sum()
```

Comment:
this is a very odd way of calling this as it should be a scalar. why do you think we should allow this way?

Reply:
not sure if I understand you. how about passing something like `level = 0` to identify the datetime column of multi index to use for comparing rolling window?

Reply:
right what I mean, is that this doesn't make rolling any more friendly to a Multi-index. we would need to add a ``level=`` kwarg that can be used internall to create the on. Would take a followup PR for that.

Reply:
ok